src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -610,6 +610,257 @@ public unsafe Vector(T[] values, int index)
             }
         }
 
+        /// <summary>
+        /// Constructs a vector from the given span.
+        /// The span must contain at least Vector'T.Count elements.
+        /// </summary>
+        public unsafe Vector(Span<T> values)
+            : this()
+        {
+            if (values == null)
+            {
+                // Match the JIT's exception type here. For perf, a NullReference is thrown instead of an ArgumentNull.
+                throw new NullReferenceException(SR.Arg_NullArgumentNullRef);
+            }
+            if (values.Length < Count)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            if (Vector.IsHardwareAccelerated)
```

Comment:
The following 230+ lines should be replaced by a single line that says:

`return Unsafe.ReadUnaligned<Vector<T>>(ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(buffer)));`

Reply:
@jkotas Please refer to [this](https://github.com/dotnet/corefx/issues/24343#issuecomment-359262863) comment in the original issue. I think I heard in the recording that a copy of the contents of the span to the vector will be done... So I based on that understanding, I modelled the entire code on lines of the constructor that accepts `T[]`... Appreciate your inputs.

Reply:
Yes, the current Vector code has a lot of complexity because of it targets everything all the way to netstandard10. 

If this API is for netcoreapp only, it can be a lot simpler.